source/_integrations/cover.mqtt.markdown

```diff
@@ -422,6 +422,108 @@ cover:
+          {{ (value | int + 1) }}
+        {% else %}
+          {% if state_attr(entity_id, "current_position") > (value | int) %}
+            {{ (value | int - 1) }}
+          {% else %}
+            {{ value }}
+          {% endif %}
+        {% endif %}
+      {% endif %}
+```
+
+{% endraw %}
+
+### Full configuration using advanced templating
+
+The example below shows a full example of how to set up a venetian blind which has a combined position and tilt topic. The blind in the example has moveable slats which tilt with a position change. In the example, it takes the blind 6% of the movement for a full rotation of the slats.
+
+Following variable might be used in `position_template`, `set_position_template`, `tilt_command_template` and `tilt_status_template`, `json_attributes_template` (only `entity_id`).
+
+- `entity_id` - The ID of the entity itself. It can be used to reference its attributes with the help of the [states](/docs/configuration/templating/#states) template function.
+- `position_open`
+- `position_closed`
+- `tilt_min`
+- `tilt_max`
+
+{% raw %}
+
+```yaml
+# Example configuration.yaml entry
+cover:
+  - platform: mqtt
+    name: "MQTT Cover"
+    command_topic: "home-assistant/cover/set"
+    state_topic: "home-assistant/cover/state"
+    position_topic: "home-assistant/cover/position"
+    set_position_topic: "home-assistant/cover/position/set"
+    tilt_command_topic: "home-assistant/cover/position/set" # same as `set_position_topic`
+    qos: 1
+    retain: false
+    payload_open:  "open"
+    payload_close: "close"
+    payload_stop:  "stop"
+    state_opening: "open"
+    state_closing: "close"
+    state_stopped: "stop"
+    position_open: 100
+    position_closed: 0
+    tilt_min: 0
+    tilt_max: 6
+    tilt_opened_value: 3
+    tilt_closed_value: 0
+    optimistic: false
+    position_template: |-
+      {% if state_attr(entity_id, "current_position") == None %}
+        {% set my_json = { "position" : value, "tilt_value" : 0 } %}
+      {% else %}
+        {% set position = state_attr(entity_id, "current_position") %}
+        {% set movement = value | int - position %}
+        {% set tilt_percent = (state_attr(entity_id, "current_tilt_position")) %}
+        {% set tilt = (tilt_percent/100*(tilt_max-tilt_min)) %}
+        {% set tilt_new = tilt + movement %}
+        {% if tilt_new > tilt_max %} {% set tilt_new = tilt_max %} {% endif %}
+        {% if tilt_new < tilt_min %} {% set tilt_new = tilt_min %} {% endif %}
+        {% set my_json = { "position" : value, "tilt_value" : tilt_new, "pos" : position, "tilt" : tilt, "tilt_percent" : tilt_percent, "mov" : movement } %}
+      {% endif %}
+      {{ my_json }}
+    tilt_command_template: |-
+      {% set position = state_attr(entity_id, "current_position") %}
+      {% set tilt = state_attr(entity_id, "current_tilt_position") %}
+      {% set movement = (tilt_position - tilt) / 100 * tilt_max %}
+      {{ (position + movement) | int }
```

Comment:
```suggestion
      {% if not state_attr(entity_id, "current_position") %}
        {
          "position" : value,
          "tilt_value" : 0
        }
      {% else %}
        {% set position = state_attr(entity_id, "current_position") %}
        {% set tilt_percent = (state_attr(entity_id, "current_tilt_position")) %}

        {% set movement = value | int - position %}
        {% set tilt = (tilt_percent / 100 * (tilt_max - tilt_min)) %}
        {% set tilt_value = min(max((tilt + movement), tilt_min), max) %}
 
        {
           "postition": value,
           "pos": position,
           "tilt": tilt,
           "tilt_value": tilt_value,
           "tilt_percent" : tilt_percent,
           "mov" : movement
        }
      {% endif %}
   tilt_command_template: >-
      {% set position = state_attr(entity_id, "current_position") %}
      {% set tilt = state_attr(entity_id, "current_tilt_position") %}
      {% set movement = (tilt_position - tilt) / 100 * tilt_max %}
      {{ position + movement }}
```

I noticed are syntax errors in this template as well. Please make sure they work.

Reply:
@m1ch Friendly ping, could you take a look at these suggestions? Beta cut-off is getting close.